Added new/extra tests for converters

a-matsuo opened the pull request:
### Summary
Added new tests or extra tests for converters of `OptimizationProblem`
I will add more tests.

### Details and comments
new tests
- [x] test_inequality_mode_integer
- [x] test_inequality_mode_continuous
- [x] test_inequality_mode_auto
- [x] test_binary_to_integer
- [x] test_optimizationproblem_to_operator
- [x] test_valid_variable_type
- [x] test_quadratic_constraints

extra tests for existing tests
- [x] test_integer_to_binary

Cryoris commented:
Hi @a-matsuo, there were some updates on the `optimization-stack` branch, could you update your branch and resolve the one conflict?

Cryoris commented on test/optimization/test_converters.py:
```diff
@@ -40,6 +65,46 @@ def test_empty_problem(self):
         qubitop, shift = conv.encode(op)
         self.assertEqual(shift, 0.0)
 
+    def test_valid_variable_type(self):
+        # validate the types of the variables for OptimizationProblemToOperator
```
If you convert this to a docstring it counts as function docstring and will not raise a pylint warning.
```suggestion
    def test_valid_variable_type(self):
        """Validate the types of the variables for OptimizationProblemToOperator."""
```

Cryoris commented on test/optimization/test_converters.py:
```diff
@@ -40,6 +65,46 @@ def test_empty_problem(self):
+            op = OptimizationProblem()
+            op.variables.add(names=['x'], types='I')
+            conv = OptimizationProblemToOperator()
+            op2 = conv.encode(op)
```
Pylint will complain about the unused variable
```suggestion
            _ = conv.encode(op)
```

Cryoris commented on qiskit/optimization/converters/integer_to_binary_converter.py:
```diff
@@ -195,6 +196,10 @@ def _substitute_int_var(self):
 
+        # TODO: add quadratic constraints
+        if self._src.quadratic_constraints.get_num() > 0:
+            warnings.warn("Quadratic constraints are currently not suppourted yet.")
```
Should this only warn here or raise an error? What do you think makes sense from user-perspective?

Cryoris commented on qiskit/optimization/converters/inequality_to_equality_converter.py:
```diff
@@ -152,6 +153,10 @@ def encode(self, op: OptimizationProblem, name: Optional[str] = None,
 
+        # TODO: add quadratic constraints
+        if self._src.quadratic_constraints.get_num() > 0:
+            warnings.warn("Quadratic constraints are currently not suppourted yet.")
```
Same, here maybe this should raise an error instead of a warning if a user tries to do a conversion we do not support yet.

a-matsuo replied:
hmm, you're right. I think raising an error is more natural. I will fix it.

a-matsuo commented:
Thank you for the comments. I added docstrings and fixed all complains of pylint. I also changed warnings of unsupported quadratic constraints to error. Could you check them please?

Cryoris commented on qiskit/optimization/converters/inequality_to_equality_converter.py:
```diff
@@ -155,7 +154,7 @@ def encode(self, op: OptimizationProblem, name: Optional[str] = None,
         # TODO: add quadratic constraints
         if self._src.quadratic_constraints.get_num() > 0:
-            warnings.warn("Quadratic constraints are currently not suppourted yet.")
+            raise QiskitOptimizationError('Quadratic constraints are currently not suppourted yet.')
```
```suggestion
            raise QiskitOptimizationError('Quadratic constraints are not yet supported.')
```

Cryoris commented on qiskit/optimization/converters/integer_to_binary_converter.py:
```diff
@@ -198,7 +190,7 @@ def _substitute_int_var(self):
         # TODO: add quadratic constraints
         if self._src.quadratic_constraints.get_num() > 0:
-            warnings.warn("Quadratic constraints are currently not suppourted yet.")
+            raise QiskitOptimizationError('Quadratic constraints are currently not suppourted yet.')
```
```suggestion
            raise QiskitOptimizationError('Quadratic constraints are not yet supported.')
```
